bap_ida_python/loader/installer.py

```diff
@@ -0,0 +1,88 @@
+#! /usr/bin/env python
+"""Loader for setting up IDA to successfully import all BAP plugins."""
+
+
+def _multi_line_string_clean(s):
+    s_split = s.split('\n')[1:]
+    for i, c in enumerate(s_split[0]):
+        if not c.isspace():
+            left_space_count = i
+            break
+    return '\n'.join(x[left_space_count:] for x in s_split)
+
+
+def get_bap_path():
+    """Return path of bap executable or None."""
+    from subprocess import check_output, CalledProcessError
+    import os
+    try:
+        bap_path = check_output(['which', 'bap']).strip()
+    except OSError:  # Cannot run 'which' command
+        print "[!] Could not determine bap path automatically."
+        print "[ ] Please enter path to bap executable (including /bap): "
+        bap_path = input('  BAP Path: ').strip()
+    except CalledProcessError:  # 'which' could not find 'bap'
+        print "[!] Unable to locate bap. Please install it and ensure that it"
+        print "        is on your path, and `bap --version` works."
+        return None
+    if os.path.isfile(bap_path) and bap_path.endswith('/bap'):
+        return bap_path
+    else:
+        print "[!] Unable to locate bap at {}".format(bap_path)
+        return None
+
+
+def install_bap_config():
```

Comment:
Once I read your points, I realized that we are in complete agreement, and that the current configuration system has issues. However, my suggestion to place it in `.bap/ida-python-config` is still valid due to the following (numbered wrt your points):
1. This is not being used as an IPC media. It is only to store the configuration. "Where is something found" is a configuration. That is all we are storing.
2. Agreed. However, since both bap and bap-ida-python are us (i.e. BAP org) itself, it isn't 3rd party. Hence using a common `.bap/` makes sense. However, since it is not the main program, we can have it in a `ida-python-config` file. Also, we could have it generate it if it doesn't exist, but not modify it if it does. In such a scenario, it'd make sense, right?
3. The information of where IDA is and where BAP is, is an integral part of the `bap-ida-python` system. Without knowing that, it would be impossible to make any calls. Hence, my proposal for the `.bap/ida-python-config`.
4. When multiple package managers are used (`opam`/source for BAP, Installer/manual for IDA Pro, `pip` for bap-ida-python), there is no package manager layer. Hence, we are left with either the OS doing some black magic for us, or us storing things in files, and this is why I chose to store it in files.

Yep, agreed, we need to discuss quite a bit, and I think it might be better if we take this offline :)